cvxpy/lin_ops/canon_backend.py

```diff
@@ -826,6 +827,290 @@ def get_param_tensor(self, shape: tuple[int, ...], parameter_id: int) \
         return {Constant.ID.value: {parameter_id: slices}}
 
 
+class NumpyCanonBackend(PythonCanonBackend):
+    """
+    Each tensor has 3 dimensions. The first one is the parameter axis, the second one is the rows
+    and the third one is the variable columns.
+
+    For example:
+    - A new variable of size n has shape (1, n, n)
+    - A new parameter of size n has shape (n, n, 1)
+    - A new constant of size n has shape (1, n, 1)
+    """
```

Comment:
This should be moved to `PythonCanonBackend`.

Reply:
Done